src/main/java/com/github/manolo8/darkbot/utils/login/LoginUtils.java

```diff
@@ -167,6 +186,16 @@ private static String replaceParameters(String params) {
         return params;
     }
 
+    private static String getLoginUrl(String in) {
+        Matcher match = LOGIN_PATTERN.matcher(in);
+        if (match.find()){
+            String login_pattern = match.group(1).replace("&amp;", "&");
+            System.out.println(login_pattern);
+            return login_pattern;
+        }
+        throw new WrongCredentialsException();
```

Comment:
i doubt credentials are wrong if you didn't try to log-in yet, find a better exception or just return a null and handle it higher up